monai/data/utils.py

```diff
@@ -243,9 +258,12 @@ def iter_patch(
     # patches which are only in the padded regions
     iter_size = tuple(s + p for s, p in zip(arr.shape, patch_size_))
 
-    for slices in iter_patch_slices(iter_size, patch_size_, start_pos_padded):
+    for position in iter_patch_position(
+        image_size=iter_size, patch_size=patch_size_, start_pos=start_pos_padded, overlap=overlap, padded=True
+    ):
+        slices = tuple(slice(s, s + p) for s, p in zip(position, patch_size_))
         # compensate original image padding
-        coords_no_pad = tuple((coord.start - p, coord.stop - p) for coord, p in zip(slices, patch_size_))
+        coords_no_pad = tuple((s - p, s) for s, p in zip(position, patch_size_))
```

Comment:
This calculation used to be in `iter_patch_slices`, things have been changed around but I'd rather this slice calculation be done in `iter_patch_position` for use cases where access to the slices themselves is needed. Otherwise places where a slice or range is needed again will have to duplicate what this line is doing which is a bit error-prone.

Reply:
I brought back `iter_patch_slices` and the calculation is being done there now. However, since this calculation it the only difference between `iter_patch_slices` and `iter_patch_position`, now `iter_patch_slices` calls `iter_patch_position` and just do this calculation.